src/machine/lib_machine.rs

```diff
@@ -228,6 +238,186 @@ impl Machine {
 
         Ok(QueryResolution::from(matches))
     }
+
+    /// NOTE: this method is entirely designed to be run as shared library code,
+    /// it will almost certainly be a footgun/resources leaks unless you are very careful!
+    pub fn start_new_query_generator(&mut self, query: String) -> QueryState {
```

Comment:
You could rebase onto #2472 if you want to build on top of that version of the interface.